### PR TITLE
Default-sample-sequencing-object

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Added functionality to allow setting of default sequencing object for a sample to be used in analyses

**No default set so first pair is displayed as default:**

no-default-set-sample

**No default set for sample so first pair is selected for an analysis by default:**

no-default-set-analysis

**When hovering on a sequencing object header the Set as Default button will be visible just for the sequencing object that the mouse is hovering over. The image is displaying all available set as default buttons just for visual reference:**

display-set-default-on-hover

**A new default sequencing object is set for the sample**

new-default-set-sample

**The default sequencing object that was set for sample is selected for an analysis by default:**

new-default-set-analysis

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
* [x] Tests added (or description of how to test) for any new features.
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/webapp/resources/js/components/sequence-files/PairedFileRenderer.jsx [hunk 1]
```diff
@@ -15,6 +15,8 @@ import { useSelector } from "react-redux";
  * @function get file processing state function
  * @param qcEntryTranslationKeys Translation keys for qc entries
  * @param displayConcatenationCheckbox Whether to display checkbox or not
+ * @function set default sequencing object for sample
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/sequence-files/PairedFileRenderer.jsx:18]: These should be `@param {function}` not `@function` . See [@function](https://jsdoc.app/tags-function.html)

File: src/main/webapp/resources/js/components/sequence-files/SequenceFileHeader.jsx [hunk 2]
```diff
@@ -14,6 +14,8 @@ import { SequenceFileHeaderOwner } from "./SequenceFileHeaderOwner";
  * @param type The type of file object (sequencingobject or genomeassembly)
  * @function remove files from sample function
  * @param displayConcatenationCheckbox Whether to display checkbox or not
+ * @function set default sequencing object for sample
+ * @param autoDefaultFirstPair the first pair in the list of pairs (if displaying paired end files)
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/sequence-files/SequenceFileHeader.jsx:18]: Same as previous, don't us `@function` here

File: src/main/webapp/resources/js/components/sequence-files/SingleEndFileRenderer.jsx [hunk 3]
```diff
@@ -16,6 +16,7 @@ import { useSelector } from "react-redux";
  * @function get file processing state function
  * @param qcEntryTranslationKeys
  * @param displayConcatenationCheckbox Whether to display checkbox or not
+ * @function set default sequencing object for sample
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/sequence-files/SingleEndFileRenderer.jsx:19]: Again!  Stop using this :)

**Review by joshsadam — CHANGES_REQUESTED** [review]
Overall I think this is actually very.  You need to update the docs. There is only one real change but there are a couple places that it needs fixing.

File: src/main/webapp/resources/js/components/sequence-files/PairedFileRenderer.jsx (hunk 1 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/sequence-files/PairedFileRenderer.jsx]: Updated in a23c555bf4b038f2ee0e43742e27ff837b87eb78

File: src/main/webapp/resources/js/components/sequence-files/SequenceFileHeader.jsx (hunk 2 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/sequence-files/SequenceFileHeader.jsx:18]: Updated in a23c555bf4b038f2ee0e43742e27ff837b87eb78

File: src/main/webapp/resources/js/components/sequence-files/SingleEndFileRenderer.jsx (hunk 3 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/sequence-files/SingleEndFileRenderer.jsx]: NEVER! Updated in a23c555bf4b038f2ee0e43742e27ff837b87eb78

File: src/main/webapp/resources/js/components/samples/components/SampleFileList.jsx [hunk 4]
```diff
@@ -80,6 +85,29 @@ export function SampleFileList() {
       .then(({ data }) => {
         notification.success({ message: data.message });
         dispatch(removeFileObjectFromSample({ fileObjectId, type }));
+        if (
+          sample.defaultSequencingObject !== null &&
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/samples/components/SampleFileList.jsx:89]: Remember you can simplify this to `sample.defaultsequencingObject?.identifier === fileObjectId`

**Review by joshsadam — CHANGES_REQUESTED** [review]
This is almost there, just a couple comments

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java [hunk 5]
```diff
@@ -194,6 +195,28 @@ public String updateSampleDetails(Long id, UpdateSampleAttributeRequest request,
...
+		try {
+			Sample sample = sampleService.read(sampleId);
+			SequencingObject sequencingObject = sequencingObjectService.readSequencingObjectForSample(sample,
+					sequencingObjectId);
+			sample.setDefaultSequencingObject(sequencingObject);
+			sampleService.update(sample);
+			return messageSource.getMessage("server.SequenceFileHeaderOwner.successfully.set.default.seq.object",
+					new Object[] {  }, locale);
+		} catch (EntityNotFoundException e) {
+			return e.getMessage();
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java]: It is now :) Updated in 4e8c30254a668636a8ec5217dd94db9910c1a389

File: src/main/webapp/resources/js/components/samples/components/SampleFileList.jsx (hunk 4 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/samples/components/SampleFileList.jsx]: Good catch. Updated in 4e8c30254a668636a8ec5217dd94db9910c1a389

File: src/main/webapp/resources/js/components/sequence-files/SequenceFileHeader.jsx (hunk 2 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/components/sequence-files/SequenceFileHeader.jsx:18]: Thanks

**Review by joshsadam — APPROVED** [review]
This looks mighty fine

**Pull request merged**